### PR TITLE
feat(rome_js_analyzer): rule `useKeyWithClickEvents`

### LINKED ISSUE #3301: `useKeyWithClickEvents`
https://github.com/rome/tools/blob/archived-js/website/src/docs/lint/rules/a11y/useKeyWithClickEvents.md

### PR DESCRIPTION
**Pull request by jk-gan**
## Summary

Closes #3301 

## Test Plan

Added various test cases

Screenshot 2022-10-09 at 5 00 20 PM
Screenshot 2022-10-09 at 5 00 17 PM

### CODE REVIEW

**Comment by jk-gan** [comment]
if possible, could you please [add hacktoberfest label to the PR](https://hacktoberfest.com/participation/#pr-mr-details)?
because otherwise it won't be count for me.

File: crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    None => Some(()),
+                }
+            }
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _attr: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+
+        Some(RuleDiagnostic::new(
```
Comment (leops) [inline crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:92]: Could we add an additional footer note to explain why these attributes are required ?

File: crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    None => Some(()),
+                }
+            }
+            JsxAnyElement::JsxSelfClosingElement(element) => {
+                element.name().ok()?.as_jsx_name()?;
+                element.find_attribute_by_name("onClick").ok()??;
+
+                match required_props
+                    .iter()
+                    .find_map(|key_event| element.find_attribute_by_name(key_event).ok().flatten())
```
Comment (xunilrj) [inline crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:80]: ```find_attribute_by_name``` is kind a expensive function, probably makes more sense do the inverse

```rust
let all_atts_present = element.attributes()
    .filter_map(|att| { 
        let name = att.as_jsx_attribute()?.name().ok()?.as_jsx_name()?.value_token().trimmed_text();
        if required_props.contains(name) { ... }
    }
);
```

**Comment by ematipico** [comment]
> if possible, could you please [add hacktoberfest label to the PR](https://hacktoberfest.com/participation/#pr-mr-details)? because otherwise it won't be count for me.

The team agreed to opt-in, we will have some issue for hacktoberfest

File: crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs (hunk 1 above)
Comment (jk-gan) [reply crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:92]: ya sure, is there a way to add additional notes? or just add more explanations on the `RuleDiagnostic`?

Comment (ematipico) [reply crates/rome_js_analyze/src/analyzers/nursery/use_key_with_click_events.rs:92]: Like this:
```rs
RuleDiagnostic::new(/* arguments */).footer_note(markup!{"Additional test"}.to_owned())
```

**Comment by jk-gan** [comment]
hmmm I think the commits are messed up after I rebased. Let me open another PR and see if Github can detect the correct file change.

**Comment by jk-gan** [comment]
#3426 this PR is able to detect the correct file change

**Pull request closed without merging**